libs/libarchfpga/src/physical_types.h

```diff
@@ -582,21 +583,15 @@ struct t_physical_tile_type {
     int width = 0;
     int height = 0;
 
-    bool**** pinloc = nullptr; /* [0..width-1][0..height-1][0..3][0..num_pins-1] */
+    std::vector<std::vector<std::vector<std::vector<bool>>>> pinloc; /* [0..width-1][0..height-1][0..3][0..num_pins-1] */
```

Comment:
Might consider using vtr::Matrix<std::vector<bool>> here (since the block width/height should be known when building the tile).

Reply:
Done. I have actually used an `NdMatrix` of dimension 3 as also the number of sides is known.